lib/shoryuken/manager.rb

```diff
@@ -19,8 +19,11 @@ def initialize(fetcher, polling_strategy)
       @polling_strategy = polling_strategy
 
       @heartbeat = Concurrent::TimerTask.new(run_now: true,
-                                             execution_interval: HEARTBEAT_INTERVAL,
-                                             timeout_interval: 60) { dispatch }
+                                             execution_interval: HEARTBEAT_INTERVAL) { @pool.post { dispatch } if @dispatching.false? }
+
+      Concurrent::TimerTask.new(execution_interval: 1) do
+        Shoryuken.logger.info "Threads: #{Thread.list.size}"
+      end.execute
```

Comment:
@waynerobinson With concurrency 10: `bundle exec bin/shoryuken -c 10 -q test -r ./test.rb`, the thread size kept consistent at 16.